### PR TITLE
feat(minifier): may add space before binaryOperator

### PR DESCRIPTION
**Pull request by mysteryven**
https://github.com/evanw/esbuild/blob/e6a8169c3a574f4c67d4cdd5f31a938b53eb7421/internal/js_printer/js_printer.go#L2920

### CODE REVIEW

**Comment by mysteryven** [comment]
```bash
Benchmark 1: ./this-pr/minifier tmp/react.js
  Time (mean ± σ):       2.6 ms ±   0.6 ms    [User: 1.6 ms, System: 0.6 ms]
  Range (min … max):     1.8 ms …   5.0 ms    447 runs

Benchmark 2: ./main/minifier tmp/react.js
  Time (mean ± σ):       2.6 ms ±   0.5 ms    [User: 1.6 ms, System: 0.6 ms]
  Range (min … max):     1.8 ms …   4.7 ms    504 runs

Summary
  './main/minifier tmp/react.js' ran
    1.00 ± 0.30 times faster than './this-pr/minifier tmp/react.js'
```

**Comment by Boshen** [comment]
Would you like to try and a few more of these according to Esbuild?

**Comment by Boshen** [comment]
You can rebase to get all the identifier related code from oxc_syntax

**Comment by mysteryven** [comment]
> Would you like to try and a few more of these according to Esbuild?

Of course, I will do more!

File: crates/oxc_minifier/src/printer/mod.rs [hunk 1]
```diff
@@ -136,6 +141,17 @@ impl Printer {
         }
     }
 
+    fn print_space_before_identifier(&mut self) {
+        let ch = unsafe { str::from_utf8_unchecked(&self.code) }.chars().last();
```
Comment (mysteryven) [inline crates/oxc_minifier/src/printer/mod.rs:145]: I am working on a new PR about [this line](https://github.com/Boshen/oxc/blob/b22c62de4f46a4230c7db9f5ea45cdd4d3c3d89f/crates/oxc_minifier/src/printer/mod.rs#L133)

I'll refactor this PR later.

**Pull request merged**